storage/database/db_migration.go

```diff
@@ -0,0 +1,123 @@
+// Modifications Copyright 2020 The klaytn Authors
+// Copyright 2015 The go-ethereum Authors
+// This file is part of go-ethereum.
+//
+// go-ethereum is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// go-ethereum is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with go-ethereum. If not, see <http://www.gnu.org/licenses/>.
+
+package database
+
+import (
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+const (
+	dbMigrationFetchNum = 500
+)
+
+// StartDBMigration migrates a DB to a different kind of DB.
+// (e.g. LevelDB -> BadgerDB, LevelDB -> DynamoDB)
+//
+// This feature uses Iterator. A src DB should have implementation of Iteratee to use this function.
+// Do not use db migration while a node is executing.
+func (dbm *databaseManager) StartDBMigration(dstdbm DBManager) error {
+	// settings for quit signal from os
+	sigQuit := make(chan os.Signal, 1)
+	signal.Notify(sigQuit,
+		syscall.SIGHUP,
+		syscall.SIGINT,
+		syscall.SIGTERM,
+		syscall.SIGQUIT)
+
+	// TODO enable for all dbs
+	srcDB := dbm.dbs[0]
+	dstDB := dstdbm.getDatabase(DBEntryType(MiscDB))
```

Comment:
Is destination always `MiscDB`?

Reply:
Why do you retrieve same `MiscDB` in different way?
- directly access `dbs` slice
- use interface method

Reply:
Currently, this feature is only provided for singleDB. It is okay to get any `DBEntryType`. I just chose the first DB.
Okay, I will change srcDB to use interface method. To be more specific, srcDB is implemented type, `databaseManager` and dstDB is interface type `DBManager`. It is impossible to directly access dbs slcie from dstDB. However, it would be better if I match how I get the DB.

Reply:
I changed PTAL